pkg/apk/implementation.go

```diff
@@ -350,18 +350,19 @@ func (a *APK) InitKeyring(ctx context.Context, keyFiles, extraKeyFiles []string)
 		eg.Go(func() error {
 			a.logger.Debugf("installing key %v", element)
 
-			// Normalize the element as a URI, so that local paths
-			// are translated into file:// URLs, allowing them to be parsed
-			// into a url.URL{}.
-			var asURI uri.URI
+			var asURL *url.URL
 			if strings.HasPrefix(element, "https://") {
-				asURI, _ = uri.Parse(element)
+				asURL, err = url.Parse(element)
```

Comment:
Can we reduce the duplication here? So you have something like:

```go
			if strings.HasPrefix(element, "https://") {
 				asURL, err = url.Parse(element)
			} else {
 				// Attempt to parse non-https elements into URI's so they are translated into
 				// file:// URLs allowing them to parse into a url.URL{}
 				asURL, err = url.Parse(string(uri.New(element)))
			}
			if err != nil {
 				return fmt.Errorf("failed to parse key as URI: %w", err)
 			}
```

It makes it easier to read.

Reply:
thanks for the suggestion, i didn't like the way my change looked either. i'll fix